server/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -851,15 +854,10 @@ private static ActionFilters setupActionFilters(List<ActionPlugin> actionPlugins
 
     public void initRestHandlers(Supplier<DiscoveryNodes> nodesInCluster) {
         List<AbstractCatAction> catActions = new ArrayList<>();
-        var restExtension = RestExtension.load(() -> new RestExtension() {
-            @Override
-            public Predicate<AbstractCatAction> getCatActionsFilter() {
-                return action -> true;
-            }
-        });
         Predicate<AbstractCatAction> catActionsFilter = restExtension.getCatActionsFilter();
+        Predicate<RestHandler> restFilter = restExtension.getActionsFilter();
```

Comment:
One thing that will be odd (although it'll work fine) is that the actions filter will return all INTERNAL and PUBLIC rest handlers (because the goal of this filtering is to drop unreachable code from memor), but the cat actions filter will return all PUBLIC cat actions only (because the goal of this filtering is to avoid confusing end users with things they can't reach). I only mention it because that tripped me up when I was first looking at this. No action needed.